### ISSUE TITLE
Duration is not being updated upon setting start and end date while using `setStartEndDate`

### ISSUE THREAD

**Issue by ghulamghousdev** [bryntum/support#8881]
[Forum post](https://forum.bryntum.com/viewtopic.php?f=44&t=28479&p=143150#p143150)

I'm using the following code to change the event start date 

```
        const currentStart = new Date(event.startDate);
        const start = new Date(currentStart.setDate(currentStart.getDate() + 1));
        event.setStartEndDate(start, event.endDate as Date);
        await eventStore.value.commit();
```

I would expect the start date to change and the end day to stay the same, making the event shorter. This does not happen though. After the EventStore.commit the event end date is recalculated to keep the length the same.

Code attached
![setStartEndDate.gif](https://forum.bryntum.com/download/file.php?id=27541)

[package.zip](https://github.com/bryntum/support/files/14748155/package.zip)

**Comment by ghulamghousdev** [comment]
Upon investigating this issue, it came to light that `setStartEndDate` method is not yet not supported in SchedulerPro. The reason for this is, SchedulerPro uses a engine for scheduling and we need to implement the setStartEndDate method at the engine level. The bug was this method shouldn't be documented in the docs.

**Comment by RytisLT** [comment]
@SergeyMaltsev how was this issue fixed in `5.6.10`? I can still reproduce the behavior with `5.6.10` and docs seems to be unchanged

```
> cat node_modules/@bryntum/schedulerpro/README.md | grep 5.6
npm install @bryntum/schedulerpro@5.6.10
yarn add @bryntum/schedulerpro@5.6.10

> cat node_modules/@bryntum/schedulerpro/schedulerpro.d.ts | grep -B 6 setStartEndDate
    /**
     * Sets the event start and end dates
     * @param {Date} start The new start date
     * @param {Date} end The new end date
     * @param {boolean} silent Pass `true` to not trigger events
     */
    setStartEndDate(start: Date, end: Date, silent?: boolean): void;
```

**Comment by ghulamghousdev** [comment]
Hi @RytisLT,

Thanks for bringing this to our intention. This is an issue with our internal decorator `@hide`. I have created a ticket here to fix it: https://github.com/bryntum/support/issues/9140
